### ISSUE TITLE
Canceled items price and tax amount should not adding in the invoice

### ISSUE THREAD

**Issue by aarya-404** [magento/magento2#33679]
### Preconditions (*)

1. Magento ver. 2.3.5-p2
2. PHP 7.3

### Steps to reproduce (*)

1. Added same product with 2 qty in the cart 
2. Used payment method Check / Money order
3. canceled 1 item qty
4. generate invoice from admin

### Expected result (*)

1. in the invoice subtotal and grand total should be for 1 qty because we canceled 1 

### Actual result (*)

1.  it's showing with 2 qty

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-November** [comment]
Verified the issue on Magento 2.4-develop branch and the issue is not reproducible:

As per the current behavior of Magento, after a customer places an Order and is in Pending Status (Before payment is received), user can edit an order or place it on hold or **cancel it entirely (User cannot cancel the Order partially for Pending State Orders**). In that case we can only issue refund for partial or full Order using Credit Memo functionality.
We can able to do partial cancellation for the Order from Admin with updated quantity (partial quantity) only after invoice is generated for partial/remaining quantity.
Also, if you want to edit/modify the Order from Admin, the original order gets cancelled and a new order is generated.
Please use these [ref docs](https://docs.magento.com/user-guide/sales/order-update.html?itm_source=merchdocs&itm_medium=search_page&itm_campaign=federated_search&itm_term=Cancel%20Order) & 
[Update Order ](https://docs.magento.com/user-guide/sales/order-update.html)  for further understanding.
Kindly recheck the issue on Magento 2.4-develop branch and provide missing steps if any if the issue is still reproducible.
Thank you

**Comment by IbrahimS2** [comment]
@engcom-November You are missing the point entirely here, the grand total should simply subtract the cancelled/refunded value, and if the product is refunded/cancelled it should not be visible in the invoices/shipping labels etc...

Long story short, grand total must be moved right at the very bottom of the order totals, as it is the final value.

**Comment by engcom-November** [comment]
@IbrahimS2, I verified the issue again on Magento 2.4-develop branch with 2 scenarios. Please find the steps followed below:

Scenario 1: Order a product with 2 Quantity - Generate Invoice for 2 Product Quantity - Refund for One Product Quantity (Credit Memo)
Front-end **Screenshots:**

Items Ordered section: Since the Order Quantity is 2 - Grand Total displayed here is for 2 Products
![image](https://user-images.githubusercontent.com/60198231/129536377-bbb75b3f-2284-45e2-8c1d-b4d9310fca2e.png)

**Invoice section** in Front-end: Grand-Total is total amount as Invoice generated for 2 Products which is generated before refund so no Refunded amount will be displayed in invoice - **No issue**
![image](https://user-images.githubusercontent.com/60198231/129535934-be4af833-81d7-4c4f-a145-7030c54c5c1a.png)

**Refund section:** Grand Total is updated as expected (For one Quantity)
![image](https://user-images.githubusercontent.com/60198231/129535823-1c03fa50-fa60-4a3e-81c6-bb8e9ecc09d6.png)

Admin **Screenshots:**
![image](https://user-images.githubusercontent.com/60198231/129536578-cc5ee95f-d6f1-4928-8f14-8044af349f81.png)
Invoiced for 2 Product Quantity: Hence the Grand Total
![image](https://user-images.githubusercontent.com/60198231/129536678-7266c7c4-16ee-42ea-9c91-5fcd08a1766c.png)
Credit Memo: Grand Total for One Product Quantity
![image](https://user-images.githubusercontent.com/60198231/129536785-60ed348a-e072-4b1c-87c8-e8e465ed4558.png)

**Comment by engcom-November** [comment]
Hi @aarya-404,
We are closing this issue as there has been no latest update on the same. Kindly reopen and provide missing steps if the issue is reproducible on 2.4-develop branch.
Thank you.
